contentcuration/contentcuration/frontend/channelList/views/Channel/ChannelItem.vue

```diff
@@ -254,7 +254,9 @@
       channelEditLink() {
         return {
           name: RouterNames.CHANNEL_EDIT,
-          query: this.$route.query,
+          query: {
+            last: this.$route.name,
```

Comment:
Could you please change this to this?
```
query: {
  ...this.$route.query,
  last: this.$route.name,
}
```
This is so that I can maintain my filters on the list if I want to edit or view the details for a channel. For instance, it would be frustrating to have applied a bunch of filters on the channel catalog, view a channel's details, then lose my filter options when I close the details. Could also make `last: this.$route.path` or something?

Reply:
I'm sorry about that - wasn't aware of the use-case you're describing.

> Could also make last: this.$route.path or something?

Whenever possible, let's stay with `this.$route.name`. If there is more last pages that need to be navigated to, we agreed to use [more explicit routes instead of more query parameters](https://learningequality.slack.com/archives/C0LE9NLCE/p1596090350285700) for now. However, in this case, it makes sense to me to keep search query parameters there.